src/Raven.Client/Documents/BulkInsert/BulkInsertOperation.cs

```diff
@@ -488,5 +519,106 @@ private string GetId(object entity)
             _generateEntityIdOnTheClient.TrySetIdentity(entity, id); //set Id property if it was null
             return id;
         }
+
+        public CountersBulkInsert CountersFor(string id)
```

Comment:
I don't like this API. In this case, we want to prioritize performance.
The common case is for us to use many counters for many documents, so we'll be creating a lot of those instances.
Better to use API such as: `IncrementAsync(string docId, string name, long delta)`